streaming/base/shared/prefix.py

```diff
@@ -108,8 +110,22 @@ def _check_and_find(streams_local: list[str], streams_remote: list[Union[str, No
     prefix_int = 0
     for prefix_int in _each_prefix_int():
         name = _get_path(prefix_int, LOCALS)
+
+        # Check if any shared memory filelocks exist for the current prefix
+        try:
+            shared_memory_exists = any(
+                os.path.exists(os.path.join(gettempdir(), _get_path(prefix_int, shm_name)))
+                for shm_name in SHM_TO_CLEAN)
+            if shared_memory_exists:
+                continue
+        except PermissionError:
```

Comment:
at which point would the permission error be thrown? when we check if the path exists?